File: swift/Sources/sdnx/Types/ReadError.swift
import Foundation

public struct ReadError: Error {
    public let message: String
    public let index: Int
    public let length: Int
    public let line: String
    public let char: Int
    
    public init(message: String, index: Int, length: Int, line: String, char: Int) {
        self.message = message
        self.index = index
        self.length = length
        self.line = line
        self.char = char
    }
}

File: dotnet/Sdnx.Core/Types/ReadError.cs
namespace Sdnx.Core
{
    public class ReadError : ParseError
    {
        public string Line { get; set; } = string.Empty;
        public int Char { get; set; }

        public ReadError()
        {
        }

        public ReadError(string message, int index, int length, string line, int ch) : base(message, index, length)
        {
            Line = line;
            Char = ch;
        }
    }
}
